### PR TITLE
Vet Clinic DB | Database Diagram #6

### PR DESCRIPTION
**Pull request by PraisesPJMT**
# Vet Clinic Database - Milestone #6 | Database Diagram
## Project Team
1. @PRATAP-KUMAR 
2. @PraisesPJMT 

## Project Description
This project involves interpreting a database diagram and implementing the SQL needed to build that database.

## Feature Description
- [X] Created `patients` table in `clinic` database.
   - Table has the following columns: `id`, `name`, and `date_of_birth`
- [X] Created `medical_histories` table in `clinic` database.
   - Table has the following columns: `id`, `admitted_at`, `status`, and `patient_id`(As Foreign Key)
- [X] Created a non clustered index for `patient_id` on `medical_histories` table
- [X] Created `treatments` table in `clinic` database.
   - Table has the following columns: `id`, `type`, and `name`
- [X] Created an `invoices` table in `clinic` database.
   - Table has the following columns: `id`, `total_amount`, `generated_amount`, `payed_at`, and `medical_history_id`(As Foreign Key)
- [X] Created a non clustered index for `medical_history_id` on `invoices` table
- [X] Created `invoice_items` table in `clinic` database.
   - Table has the following columns: `id`, `unit_price`, `total_price`, `total_price`, `quantity`, `invoice_id`,(As Foreign Key) and `treatment_id`(As Foreign Key)
- [X] Created a non clustered index for `invoice_id` on `invoice_items` table

Happy Code Reviewing :+1: :slightly_smiling_face:

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+*/
+
+CREATE TABLE invoice_items(
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    unit_price DECIMAL,
+    total_price DECIMAL,
+    quantity INT,
+    invoice_id INT,
+    treatment_id INT,
+
```
Comment (ShadyShawkat) [inline schema_based_on_diagram.sql:82]: - [x] To represent the relation between the `invoice_items` and `treatments` tables according to the [given diagram](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/images/clinic_diagram.png), kindly add the `treatment_id` column as a foreign key along with its index.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (PraisesPJMT) [reply schema_based_on_diagram.sql:82]: Thanks for the recommendation! We just implemented the required changes!

**Comment by PraisesPJMT** [comment]
## Implemented Recommended Changes
Thanks so much, @ShadyShawkat for the review.
The following changes were made based on the recommended changes.
- [X] [Add the `treatment_id` column on `invoice_items` table as a foreign key to represent the relation between the `invoice_items` and `treatments` tables](https://github.com/PraisesPJMT/Veterinary-Clinic-DB/pull/5/commits/dfe9f9491adeb8d0df60c81a3ee0f7d9bb211386)

Happy Code Reviewing :+1: :slightly_smiling_face:

**Review by ShahierNashaat — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PraisesPJMT** [comment]
Thanks @ShahierNashaat for the amazing review!

Happy Coding :+1:

**Pull request merged**